Mathlib/Algebra/Lie/Weights/Killing.lean

```diff
@@ -557,6 +557,12 @@ lemma finrank_rootSpace_eq_one (α : Weight K H L) (hα : α.IsNonZero) :
   replace hn : -2 = (n : ℤ) := by norm_cast at hn
   omega
 
+/-- The embedded `sl₂` associated to a root. -/
+noncomputable def sl2SubalgebraOfRoot {α : Weight K H L} (hα : α.IsNonZero) :
```

Comment:
I think for this to be useful I should add a lemma along the following lines here:
```lean
lemma mem_sl2SubalgebraOfRoot_iff {α : Weight K H L} (hα : α.IsNonZero) :
    ∃ h e f : L, IsSl2Triple h e f ∧ e ∈ rootSpace H α ∧ f ∈ rootSpace H (- α) ∧ ∀ x : L,
    x ∈ sl2SubalgebraOfRoot hα ↔ ∃ c₁ c₂ c₃ : K, x = c₁ • e + c₂ • f + c₃ • ⁅e, f⁆ := by
  choose h e f t ht using exists_isSl2Triple_of_weight_isNonZero hα
  refine ⟨h, e, f, t, ht.1, ht.2, fun x ↦ ⟨fun hx ↦ ?_, fun ⟨c₁, c₂, c₃, hx⟩ ↦ ?_⟩⟩
  · sorry
  · sorry
```